### ISSUE TITLE
Allow passing a simple date on the command line

### ISSUE THREAD

**Issue by Mossop** [Mossop/glowmarkt-rs#4]
It's probably common to want to grab readings for specific days while at the moment you have to pass a full date time to the command line. Being able to pass say `2023-11-01` and having that automatically recognized as `2023-11-01T00:00:00` would be helpful.

One way to approach this would be to attempt to parse as a [`Date`](https://docs.rs/time/latest/time/struct.Date.html) first and then try `OffsetDateTime` if that fails. Another would be to just throw a load of regular expressions at the problem.

**Comment by jnioche** [comment]
Had a try yesterday but no luck yet. Am not quite there with my Rust skills yet, will try again once I've learned more.

BTW have switched to Octopus and just came across [octopower](https://crates.io/crates/octopower), which does overlap with the Glowmarkt API. Will have a look at that too

**Comment by Mossop** [comment]
Yeah I had heard that if Octopus is your energy provider then they have their own API. I don't use them unfortunately.
